toxcore/net_crypto.h

```diff
@@ -20,6 +20,7 @@
 #include "logger.h"
 #include "mem.h"
 #include "mono_time.h"
+#include "net_profile.h"
```

Comment:
here too ... i feel like we should forward declare it in some public places.

Reply:
Forward decls are more verbose in C:
```c
#ifndef MY_TYPE_DEFINED
#define MY_TYPE_DEFINED
typedef struct My_Type My_Type;
#endif /* MY_TYPE_DEFINED */
```

I'm not convinced we should do more of that outside public API headers.